p2p/querier.go

```diff
@@ -0,0 +1,62 @@
+package p2p
+
+import (
+	"context"
+	"time"
+
+	celestiatypes "github.com/celestiaorg/celestia-app/x/qgb/types"
+	"github.com/celestiaorg/orchestrator-relayer/types"
+)
+
+// QuerierI queries the P2P network for confirms.
+// Note: for now, we will be querying the P2P network directly without storing anything in the store.
+// this would make it easier to have a simple working version. Then, we can iterate from there on.
+type QuerierI interface {
+	// QueryTwoThirdsDataCommitmentConfirms queries the p2p network for data commitments signed by more
+	// than 2/3s of the last validator set.
+	// It takes a timeout to wait for validators to sign and broadcast their signatures.
+	// Should return an empty slice and no error if it couldn't find enough signatures.
+	QueryTwoThirdsDataCommitmentConfirms(
+		ctx context.Context,
+		timeout time.Duration,
+		dc celestiatypes.DataCommitment,
+	) ([]types.DataCommitmentConfirm, error)
+
+	// QueryTwoThirdsValsetConfirms queries the p2p network for valsets signed by more
+	// than 2/3s of the last validator set.
+	// It takes a timeout to wait for validators to sign and broadcast their signatures.
+	// Should return an empty slice and no error if it couldn't find enough signatures.
+	QueryTwoThirdsValsetConfirms(
+		ctx context.Context,
+		timeout time.Duration,
+		valset celestiatypes.Valset,
+	) ([]types.ValsetConfirm, error)
+
+	// QueryValsetConfirmByOrchestratorAddress get the valset confirm having nonce `nonce`
+	// and signed by the orchestrator whose address is `address`.
+	QueryValsetConfirmByOrchestratorAddress(
+		ctx context.Context,
+		nonce uint64,
+		address string,
+	) (*types.ValsetConfirm, error)
+
+	// QueryDataCommitmentConfirmByOrchestratorAddress get the data commitment confirm having nonce `nonce`
+	// and signed by the orchestrator whose address is `address`.
+	QueryDataCommitmentConfirmByOrchestratorAddress(
+		ctx context.Context,
+		nonce uint64,
+		address string,
+	) (*types.DataCommitmentConfirm, error)
+
+	// QueryDataCommitmentConfirms get all the data commitment confirms in store for a certain nonce.
+	QueryDataCommitmentConfirms(
+		ctx context.Context,
+		nonce uint64,
+	) ([]types.DataCommitmentConfirm, error)
+
+	// QueryValsetConfirms get all the valset confirms in store for a certain nonce.
+	QueryValsetConfirms(
+		ctx context.Context,
+		nonce uint64,
+	) ([]types.ValsetConfirm, error)
+}
```

Comment:
I'm not sure if this is a good starting point for the interface we're looking for imho. For example, We are not planning on having multiple implementations of `QueryForTwoThirds`, so I'm not sure atm if it makes sense to have it. It seems like that, and many of the others here, can simply be a function or a method.

Reply:
> If these things are just for testing purposes

Yes, 100% they are. As stated here https://github.com/celestiaorg/orchestrator-relayer/pull/50#discussion_r1060298133, I would love to see some examples of mocks to improve these big interfaces.

Reply:
I think we should just not mock anything, and use the testnode code, which should be pretty easy to setup a testsuite for, and can have super short block times. If we really wanted to get clever, we can store some state/blocks as test data, and then boot up that "frozen" chain. 

Alternatively, we could generate mocks for all of the core/app rpc endpoints needed, but that's only marginally better than the other form of mocks and not be worth the effort